src/scripts/core/models/background.ts

```diff
@@ -67,66 +83,54 @@ export class Background extends Container {
+        // gsap.from(this._ship, {
+        //     x: this._viewWidth + this._ship.width,
+        //     duration: 3
+        // })
     }
 
     private initWave(sprite: Sprite, width: number, height: number): void {
         sprite.width = width;
         sprite.height = height;
     }
 
-    private backgroundSpriteTweenManager(waveHeightMultiplier: number): void {
-        const farTime = Math.floor(Math.random() * 5) + 3;
-        const midTime = Math.floor(Math.random() * 5) + 3;
-        const closeTime = Math.floor(Math.random() * 5) + 3;
+    public setWaveHeight(farWaveHeightMod: number, midWaveHeightMod: number, closeWaveHeightMod: number, duration: number): void {
+        this.resetSplashTimeline(farWaveHeightMod, midWaveHeightMod ,closeWaveHeightMod, duration);
+    }
 
-        if (this._splash) { this._splash.kill(); }
-        this._splash = gsap.timeline();
+    private addWavesToTimeline(farWaveHeightMod: number, midWaveHeightMod: number, closeWaveHeightMod: number, duration: number): void {
+        const farTime = Math.floor(Math.random() * duration) + 4;
+        const midTime = Math.floor(Math.random() * duration) + 3;
+        const closeTime = Math.floor(Math.random() * duration) + 2;
 
-        this._splash.to(this._farWave, {
-            y: 0 - this._farWave.height * waveHeightMultiplier,
-            duration: farTime,
+        this._splash = gsap.timeline({
             repeat: -1,
             yoyo: true
-        }, 0);
+        });
 
-        this._splash.to(this._midWaveContainer, {
-            y: 0 - this._ship.height * waveHeightMultiplier,
+        this._splash.to(this._farWave, {
+            y: this._waveStartingPosition - (this._closeWave.height * farWaveHeightMod),
+            duration: farTime,
+        }, 0)
+        .to(this._midWaveContainer, {
+            y: this._midWaveContainer.y - (this._midWave.height * midWaveHeightMod),
             duration: midTime,
-            repeat: -1,
-            yoyo: true
+        }, 0)
+        .to(this._closeWave, {
+            y: this._waveStartingPosition - (this._closeWave.height * closeWaveHeightMod),
+            duration: closeTime,
         }, 0);
 
-        this._splash.fromTo(this._ship, {
-            angle: -20,
-        }, {
-            angle: 20,
-            duration: 5,
-            repeat: -1,
-            yoyo: true
-        }, 0);
+    }
 
-        this._splash.to(this._closeWave, {
-            y: 0 - this._closeWave.height * waveHeightMultiplier,
-            duration: closeTime,
-            repeat: -1,
-            yoyo: true
-        }, 0);
+    private resetSplashTimeline(farWaveHeightMod: number, midWaveHeightMod: number, closeWaveHeightMod: number, duration: number): void {
+        this._splash.then( () => {
+            this._splash.yoyo(false);
+        })
+        .then(() => {
+            this.addWavesToTimeline(farWaveHeightMod, midWaveHeightMod, closeWaveHeightMod, duration);
+        });
```

Comment:
As mentioned I hadn't realised you could use `.then` from a Timeline so this was new to me. Had a little look though and it is just a Promise implementation of the onComplete approach. So the 2 should be interchangable.

With the chaining here you're not getting any thing from the second `.then` and can instead put it all in one like so:
```ts
this._splash.then(() => {
    this.splash.yoyo(false);
   this.addWavesToTimeline(etc...);
});
```

The ability to chain `.then` together is for when you have other asynchronous code you want to run inside the `.then` before the next `.then` should happen i.e.

```ts
this._splash.then(() => {
    return requestSomeDataFromServer();
}).then((response) => {
    // this would happen after receiving the response from the server
    console.log(response);
});
```

Reply:
Aha! That makes more sense now
I'll make the change